src/core/zowe/core_for_zowe_sdk/profile_manager.py

```diff
@@ -58,260 +71,169 @@ def config_appname(self) -> str:
 
     @property
     def config_dir(self) -> Union[str, None]:
-        """Returns the folder path to where the Zowe z/OSMF Team Profile Config files are located."""
-        return self._config_dir
+        """Returns the folder path to where the Zowe z/OSMF Team Project Config files are located."""
+        return self.project_config.location
 
     @config_dir.setter
     def config_dir(self, dirname: str) -> None:
-        """Set directory/folder path to where Zowe z/OSMF Team Profile Config files are located"""
-        if os.path.isdir(dirname):
-            self._config_dir = dirname
-        else:
-            raise FileNotFoundError(f"given path {dirname} is not valid")
+        """
+        Set directory/folder path to where Zowe z/OSMF Team Project Config files are located
+        """
+        self.project_config.location = dirname
+        self.project_user_config.location = dirname
+
+    @property
+    def user_config_dir(self) -> Union[str, None]:
+        """Returns the folder path to where the Zowe z/OSMF User Project Config files are located."""
+        return self.project_user_config.location
+
+    @user_config_dir.setter
+    def user_config_dir(self, dirname: str) -> None:
+        """Set directory/folder path to where Zowe z/OSMF User Project Config files are located"""
+        self.project_user_config.location = dirname
 
     @property
     def config_filename(self) -> str:
-        """Return the filename for Zowe z/OSMF Team Profile Config"""
-        return self._config_filename
+        """Return the filename for Zowe z/OSMF Team Project Config"""
+        return self.project_config.filename
 
     @property
     def config_filepath(self) -> Union[str, None]:
-        """Get the full Zowe z/OSMF Team Config filepath"""
-        return self._config_filepath
-
-    def autodiscover_config_dir(self) -> None:
-        """Autodiscover Zowe z/OSMF Team Profile Config files by going up the path from
-        current working directory"""
-
-        current_dir = os.getcwd()
-
-        while self._config_dir is None:
-            path = os.path.join(current_dir, self._config_filename)
+        """Get the full Zowe z/OSMF Team Project Config filepath"""
+        return self.project_config.filepath
 
-            if os.path.isfile(path):
-                self._config_dir = current_dir
+    @staticmethod
+    def get_profile(
+        cfg: ConfigFile,
+        profile_name: Union[str, None],
+        profile_type: Union[str, None],
+        config_type: str,
```

Comment:
We cannot use `str | None` because this syntax was introduced only in Python 3.10 ... however, yes, Optional is a better choice, I will update my code